TST: Array API tests fail with "Interoperation with NumPy requires python >= 3.12"

Seen in devdeps. What does this mean? Do we have to drop Python 3.11 now?

`FAILED astropy/units/tests/test_quantity_array_methods.py::test_array_api_init - TypeError: Interoperation with NumPy requires python >= 3.12. Please upgrade.`

`FAILED astropy/units/tests/test_units.py::test_array_api_strict_arrays - TypeError: Interoperation with NumPy requires python >= 3.12. Please upgrade.`

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -118,6 +118,7 @@ test_all = [
     "skyfield>=1.20",
     "sgp4>=2.3",
     "array-api-strict>=1.0",
+    "array-api-strict<2.4;python_version<'3.12'",  # PYTHON_LT_3_12
 ]
 typing = [
     "pandas-stubs>=2.0",

--- a/tox.ini
+++ b/tox.ini
@@ -100,6 +100,7 @@ deps =
     devdeps-!noscipy: skyfield>=1.20
     devdeps-!noscipy: sgp4>=2.3
     devdeps-!noscipy: array-api-strict>=1.0
+    py311-devdeps-!noscipy: array-api-strict<2.4
 
 # The following indicates which [project.optional-dependencies] from pyproject.toml will be installed.
 # test_all does not work here due to upstream bug https://github.com/tox-dev/tox/issues/3433

